consensus/hotstuff/pacemaker/pacemaker_test.go

```diff
@@ -403,29 +399,48 @@ func (s *ActivePaceMakerTestSuite) Test_Initialization() {
 	// Verify that WithTCs still works correctly if no TCs are given:
 	// the list of TCs is empty or all contained TCs are nil
 	s.Run("Only nil TCs", func() {
-		pm, err := New(timeout.NewController(s.timeoutConf), s.notifier, s.persist, WithTCs())
+		pm, err := New(timeout.NewController(s.timeoutConf), NoProposalDelay(), s.notifier, s.persist, WithTCs())
 		require.NoError(s.T(), err)
 		require.Equal(s.T(), s.initialView, pm.CurView())
 
-		pm, err = New(timeout.NewController(s.timeoutConf), s.notifier, s.persist, WithTCs(nil, nil, nil))
+		pm, err = New(timeout.NewController(s.timeoutConf), NoProposalDelay(), s.notifier, s.persist, WithTCs(nil, nil, nil))
 		require.NoError(s.T(), err)
 		require.Equal(s.T(), s.initialView, pm.CurView())
 	})
 
 	// Verify that WithQCs still works correctly if no QCs are given:
 	// the list of QCs is empty or all contained QCs are nil
 	s.Run("Only nil QCs", func() {
-		pm, err := New(timeout.NewController(s.timeoutConf), s.notifier, s.persist, WithQCs())
+		pm, err := New(timeout.NewController(s.timeoutConf), NoProposalDelay(), s.notifier, s.persist, WithQCs())
 		require.NoError(s.T(), err)
 		require.Equal(s.T(), s.initialView, pm.CurView())
 
-		pm, err = New(timeout.NewController(s.timeoutConf), s.notifier, s.persist, WithQCs(nil, nil, nil))
+		pm, err = New(timeout.NewController(s.timeoutConf), NoProposalDelay(), s.notifier, s.persist, WithQCs(nil, nil, nil))
 		require.NoError(s.T(), err)
 		require.Equal(s.T(), s.initialView, pm.CurView())
 	})
 
 }
 
+type dynamicProposalDurationProvider struct {
+	dur time.Duration
+}
+
+func (p *dynamicProposalDurationProvider) ProposalDuration() time.Duration {
+	return p.dur
+}
+
```

Comment:
lets make it obvious that we implement a specific interface

Reply:
I think alex has done this in his revision